pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -40,22 +39,23 @@
 import org.apache.pinot.core.operator.filter.TextMatchFilterOperator;
 import org.apache.pinot.core.operator.filter.predicate.PredicateEvaluator;
 import org.apache.pinot.core.operator.filter.predicate.PredicateEvaluatorProvider;
+import org.apache.pinot.core.query.request.context.QueryContext;
 import org.apache.pinot.core.segment.index.readers.NullValueVectorReader;
 
 
 public class FilterPlanNode implements PlanNode {
-  private final BrokerRequest _brokerRequest;
-  private final IndexSegment _segment;
+  private final IndexSegment _indexSegment;
+  private final QueryContext _queryContext;
 
-  public FilterPlanNode(IndexSegment segment, BrokerRequest brokerRequest) {
-    _segment = segment;
-    _brokerRequest = brokerRequest;
+  public FilterPlanNode(IndexSegment indexSegment, QueryContext queryContext) {
+    _indexSegment = indexSegment;
+    _queryContext = queryContext;
   }
 
   @Override
   public BaseFilterOperator run() {
-    FilterQueryTree rootFilterNode = RequestUtils.generateFilterQueryTree(_brokerRequest);
-    return constructPhysicalOperator(rootFilterNode, _segment, _brokerRequest.getDebugOptions());
+    FilterQueryTree rootFilterNode = RequestUtils.generateFilterQueryTree(_queryContext.getBrokerRequest());
```

Comment:
So, this will go away in future PRs where instead of generating FilterQueryTree per FilterPlanNode (per segment), we will just use the already built FilterContext from QueryContext?

Reply:
Yes, this is removed in #5584